sbi/inference/snpe/snpe_base.py

```diff
@@ -1,97 +1,96 @@
 from abc import ABC, abstractmethod
 from copy import deepcopy
-from typing import Callable, Optional, Tuple, Dict, Union
-import warnings
+from typing import Callable, Dict, Optional, Tuple, Union, cast
+from warnings import warn
 
 import numpy as np
 import torch
-from torch import Tensor, nn, optim, zeros, ones
+from torch import Tensor, nn, ones, optim, zeros
 from torch.nn.utils import clip_grad_norm_
 from torch.utils import data
 from torch.utils.data.sampler import SubsetRandomSampler
 from torch.utils.tensorboard import SummaryWriter
 
 from sbi.inference import NeuralInference
 from sbi.inference.posterior import NeuralPosterior
-from sbi.types import ScalarFloat, OneOrMore
+from sbi.types import OneOrMore, ScalarFloat
 import sbi.utils as utils
 from sbi.utils import Standardize, handle_invalid_x, warn_on_invalid_x
+from sbi.utils.torchutils import get_default_device
 
 
 class PosteriorEstimator(NeuralInference, ABC):
     def __init__(
         self,
-        simulator: Callable,
         prior,
+        simulator: Callable,
         x_shape: Optional[torch.Size] = None,
-        density_estimator: Optional[nn.Module] = None,
-        calibration_kernel: Optional[Callable] = None,
+        num_workers: int = 1,
+        simulation_batch_size: int = 1,
+        density_estimator: Union[str, nn.Module] = "maf",
         z_score_x: bool = True,
         z_score_min_std: float = 1e-7,
-        simulation_batch_size: Optional[int] = 1,
-        retrain_from_scratch_each_round: bool = False,
-        discard_prior_samples: bool = False,
-        device: Optional[torch.device] = None,
         sample_with_mcmc: bool = False,
         mcmc_method: str = "slice_np",
-        num_workers: int = 1,
-        summary_writer: Optional[SummaryWriter] = None,
+        calibration_kernel: Optional[Callable] = None,
+        retrain_from_scratch_each_round: bool = False,
+        discard_prior_samples: bool = False,
         skip_input_checks: bool = False,
+        exclude_invalid_x: bool = True,
```

Comment:
Did we change the policy that this is `False` by default?

Reply:
Yes, I discussed with Michael and we were thinking where to place it. If it is `False`, then because 'there will be blood' it should be early and prominent in the argument list. But if it is `True`, then no bad things will happen and we can put it down with the more arcane arguments. We felt that having high likelihood of exploding was not a good default even for the 'split interface'. 

We also thought that it will be better a training argument than instantiation one (this will be handled by @michaeldeistler in an upcoming PR, along other such not-instantiation-rather-training-time changes).

Of course, we may have been blinded and would be willing to reconsider both, we just did not want to distract you from your work today.

Reply:
Thanks, I think that makes sense!